### ISSUE TITLE
Low loss but still getting poor segmentation

### ISSUE THREAD

**Issue by MBronars** [lmb-freiburg/Unet-Segmentation#47]
I am fine-tuning a model to detect neural stem cells.  The cell shape is not very consistent from picture to picture and the cells will often go out of focus.  The loss and intersection over union are both pretty good (.15 and .7 respectively) but the segmentation is very poor (under .1).  When I try to segment using the trained model it either picks up on nothing or partially recognizes a few cells.  I am confused why the performance is so good on the other two metrics and yet it still fails to segment.  Do you have any suggestions for how I can get better segmentation results?  Thanks!

**Comment by MBronars** [comment]
Here are a few images:

![Finetuning test](https://user-images.githubusercontent.com/35738375/61567629-142d1c00-aa35-11e9-8ff2-0811e681b7ad.jpg)
These are the settings I used for fine-tuning the current model.

![Test7 Data](https://user-images.githubusercontent.com/35738375/61567673-450d5100-aa35-11e9-9a6d-801be461824d.JPG)
Here are the graphs from the training.

![test7 simple cell](https://user-images.githubusercontent.com/35738375/61567682-50f91300-aa35-11e9-97a5-d247fd710eea.JPG)
Here is a validation image of a simple cell, works pretty well.

![test7 hard cell](https://user-images.githubusercontent.com/35738375/61567710-6706d380-aa35-11e9-89d2-a3d18aeee49e.JPG)
![validation image T7](https://user-images.githubusercontent.com/35738375/61567712-68d09700-aa35-11e9-9352-aa68e4cad8f3.JPG)
These are validation images of more difficult cells.  Does not have good segmentation.

![Training Image 7](https://user-images.githubusercontent.com/35738375/61567756-90bffa80-aa35-11e9-9752-4aea4d582af4.JPG)
Segmentation results are much better for this one, but this is one of the training images, not validation.

**Comment by ThorstenFalk** [comment]
Even on the training image I would call the segmentation at best poor... Your training images seem to be quite diverse, looking at the repetitive pattern in the loss curve varying by more than factor 20. 2000 iterations are not very many especially with out-of-focus objects that must be rejected. Maybe the network just needs a little more time? The checkerboard-like segmentation also indicates that the model is not yet stabilized.

**Comment by MBronars** [comment]
I will try it again with more iterations and see how it goes.  Would it help to re-finetune this partially trained model or is it best to always start from your pretrained 2d_cell_net_v0 model?

**Comment by ThorstenFalk** [comment]
You can safely continue from your model.
